From 75d11f2d89eb78593496e1dcc65fdb50345149f8 Mon Sep 17 00:00:00 2001
From: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Date: Wed, 1 Jul 2020 04:47:27 +0530
Subject: [PATCH 1407/1852] spi: spi-xilinx: Fix incorrect variable type

commit e536147048149bc6dab9f04e962d7f618a4a242a from
https://github.com/Xilinx/linux-xlnx.git

Some variables with incorrect type were passed to "of_property_read_u32"
API, "of_property_read_u32" API was expecting an "u32 *" but the formal
parameter that was passed was of type "int *". Fixed the issue by
changing the variable types from "int" to "u32".

Addresses-Coverity: "incompatible_param"
Signed-off-by: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Acked-by: Naga Sureshkumar Relli <naga.sureshkumar.relli@xilinx.com>
Signed-off-by: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/spi/spi-xilinx.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/spi/spi-xilinx.c b/drivers/spi/spi-xilinx.c
index 9acfb6a65078..33818b7b09a1 100644
--- a/drivers/spi/spi-xilinx.c
+++ b/drivers/spi/spi-xilinx.c
@@ -642,7 +642,8 @@ static int xilinx_spi_probe(struct platform_device *pdev)
 {
 	struct xilinx_spi *xspi;
 	struct resource *res;
-	int ret, num_cs = 0, bits_per_word = 8;
+	int ret;
+	u32 num_cs = 0, bits_per_word = 8;
 	u32 cs_num;
 	struct spi_master *master;
 	struct device_node *nc;
-- 
2.31.1

